lib/XTR/ReportScheduleTask.php

```diff
@@ -126,51 +126,63 @@ private function runReportSchedule()
+                    $result = $zip->open($zipName, \ZipArchive::CREATE | \ZipArchive::OVERWRITE);
 
-                // Create a ZIP file and add our temporary file
-                $zipName = $this->config->getSetting('LIBRARY_LOCATION') . 'temp/reportschedule.json.zip';
-                $zip = new \ZipArchive();
-                $result = $zip->open($zipName, \ZipArchive::CREATE | \ZipArchive::OVERWRITE);
-                if ($result !== true)
-                    throw new \InvalidArgumentException(__('Can\'t create ZIP. Error Code: %s', $result));
+                    if ($result !== true) {
+                        $this->log->error(__('Can\'t create ZIP. Error Code: %s', $result));
+                        continue;
+                    }
 
-                $zip->addFile($fileName, 'reportschedule.json');
-                $zip->close();
+                    $zip->addFile($fileName, 'reportschedule.json');
+                    $zip->close();
 
-                // Remove the JSON file
-                unlink($fileName);
+                    // Remove the JSON file
+                    unlink($fileName);
 
-                $runDateTimestamp = $this->date->parse()->format('U');
+                    $runDateTimestamp = $this->date->parse()->format('U');
 
-                // Upload to the library
-                $media = $this->mediaFactory->create(__('reportschedule_' . $reportSchedule->reportScheduleId . '_' . $runDateTimestamp ), 'reportschedule.json.zip', 'savedreport', $reportSchedule->userId);
-                $media->save();
+                    // Upload to the library
+                    $media = $this->mediaFactory->create(__('reportschedule_' . $reportSchedule->reportScheduleId . '_' . $runDateTimestamp ), 'reportschedule.json.zip', 'savedreport', $reportSchedule->userId);
+                    $media->save();
 
-                // Save Saved report
-                $savedReport = $this->savedReportFactory->create($saveAs, $reportSchedule->reportScheduleId, $media->mediaId, time(), $reportSchedule->userId);
-                $savedReport->save();
+                    // Save Saved report
+                    $savedReport = $this->savedReportFactory->create($saveAs, $reportSchedule->reportScheduleId, $media->mediaId, time(), $reportSchedule->userId);
+                    $savedReport->save();
 
-                $this->createPdfAndNotification($reportSchedule, $savedReport, $media);
+                    $this->createPdfAndNotification($reportSchedule, $savedReport, $media);
+
+                } catch (\Exception $error) {
+                    $this->log->error('Error: ' . $error->getMessage());
+                    continue;
+                }
 
                 // Add the last savedreport in Report Schedule
                 $this->log->debug('Last savedReportId in Report Schedule: '. $savedReport->savedReportId);
                 $rs->lastSavedReportId = $savedReport->savedReportId;
+                $rs->isActive = 1;
```

Comment:
We set it to 0 at the beginning and set it to 1 at the end if its failed, but doesn't the `continue` in the catch skip this step?

I think we need to remove the continue from the catch?

It also makes more logical sense to me to leave isActive at the top, and to set it to 0 in the catch (so that if we fail, we set it to 0)?

Reply:
Sorry, I see the `continue` is necessary - but it still seems strange to me that we're saving it before we run it, and again after.

Can't we save one time at the end instead?